src/view.c

```diff
@@ -1792,6 +1792,46 @@ view_snap_to_region(struct view *view, struct region *region,
 	view_apply_region_geometry(view);
 }
 
+void
+view_move_to_output(struct view *view, struct output *output)
+{
+	assert(view);
+	if (view->fullscreen) {
+		return;
+	}
+
+	view_invalidate_last_layout_geometry(view);
+	view_set_output(view, output);
+	if (view_is_floating(view)) {
+		struct wlr_box output_area = output_usable_area_in_layout_coords(output);
+		view->pending.x = output_area.x;
+		view->pending.y = output_area.y;
+		view_place_initial(view, /* allow_cursor */ false);
+	} else if (view->maximized != VIEW_AXIS_NONE) {
+		view_apply_maximized_geometry(view);
+	} else if (view->tiled) {
+		view_apply_tiled_geometry(view);
+	} else if (view->tiled_region) {
+		struct region *region = regions_from_name(view->tiled_region->name, output);
+		view_snap_to_region(view, region, /*store_natural_geometry*/ false);
+	}
+}
+
+void
+view_fit_to_output(struct view *view)
```

Comment:
This looks similar to `view_constrain_size_to_that_of_usable_area()`. Are we duplicating?

Reply:
shame on me, I was ;) Much better now with re-using that method. I also found a small bug while testing, see the last commit.

Reply:
PS: Just to be sure, `view_constrain_size_to_that_of_usable_area()` won't explode for full screen windows?

Reply:
@johanmalm  I'm thinking that this wouldn't hurt, but dunno if it is really needed:

```diff
diff --git a/src/view.c b/src/view.c
index 4849318..b3f5c90 100644
--- a/src/view.c
+++ b/src/view.c
@@ -747,7 +747,7 @@ view_place_initial(struct view *view, bool allow_cursor)
 void
 view_constrain_size_to_that_of_usable_area(struct view *view)
 {
-       if (!view || !view->output) {
+       if (!view || !view->output || view->fullscreen) {
                return;
        }
 

```

Reply:
^ merged via
- #1464